pkg/genericapiserver/server/storage_factory.go

```diff
@@ -30,6 +30,15 @@ import (
 	"github.com/golang/glog"
 )
 
+// Backend describes the storage servers, the information here should be enough
+// for health validations
+type Backend struct {
+	Server string
+	// TLS credentials
```

Comment:
> these will be empty if it is not an HTTPS connection.

The client certificate is only required when the server enables client-cert-auth.

There are already many places use `CertFile` and `KeyFile` around, like `storagebackend.Config`, so I just leave it as it is..